src/hotspot/os/windows/os_windows.cpp

```diff
@@ -4413,8 +4386,38 @@ bool os::message_box(const char* title, const char* message) {
   return result == IDYES;
 }
 
+HMODULE GetHotspotModuleHandle() {
+    HMODULE hModule = NULL;
+    GetModuleHandleEx(
+        GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+        (LPCTSTR) &GetHotspotModuleHandle,
+        &hModule
+    );
+    return hModule;
+}
+
+static void windows_at_exit_callback() {
+    if (ForceTimeHighResolution) {
+      timeEndPeriod(1L);
+    }
+#if defined(USE_VECTORED_EXCEPTION_HANDLING)
+    if (topLevelVectoredExceptionHandler != nullptr) {
+      RemoveVectoredExceptionHandler(topLevelVectoredExceptionHandler);
+      topLevelVectoredExceptionHandler = nullptr;
+    }
+#endif
+}
+
 // This is called _before_ the global arguments have been parsed
 void os::init(void) {
+    vm_lib_handle = GetHotspotModuleHandle();
+    if (ForceTimeHighResolution) {
+      timeBeginPeriod(1L);
+    }
+    WindowsDbgHelp::pre_initialize();
+    SymbolEngine::pre_initialize();
```

Comment:
As David has pointed out, this is too late. There is a reason we added this to DllMain - we need this initialized very early in case we crash before this point. We still will want useful hs-err files.